src/json.hpp

```diff
@@ -144,6 +203,23 @@ struct DecimalSeparator : std::numpunct<char>
 
 }
 
+// taken from ranges-v3
+// TODO add doc
+template <typename T>
+struct __static_const
+{
+  static constexpr T value{};
+};
+
+template <typename T>
+constexpr T __static_const<T>::value;
```

Comment:
According to [cppreference](http://en.cppreference.com/w/cpp/language/identifiers) : "the identifiers with a double underscore anywhere are reserved"
 
I'll rename it